### PR TITLE
[proc] handle OS signals gracefully

### PR DESCRIPTION
**Pull request by simonlingoogle**
This PR enhance OTNS to handle OS signals gracefully:

- Exit gracefully on signals: TERM, QUIT, INT, HUP
- Ignore signals: ALRM, CHLD
- Add a Python unit test to verify signals are handled properly.

### CODE REVIEW

File: pylibs/unittests/test_signals.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+import signal
```
Comment (Irving-cl) [inline pylibs/unittests/test_signals.py:27]: nit: add a line between copyright and imports?

File: pylibs/unittests/test_signals.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    def _send_signal(self, delay: float, sig: int):
+        print(f"sleep {delay} ...")
+        time.sleep(delay)
+        print(f'sending signal {sig}')
+        self.ns._otns.send_signal(sig)
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (Irving-cl) [inline pylibs/unittests/test_signals.py:106]: nit: add an empty line at the end?

File: pylibs/unittests/test_signals.py (hunk 1 above)
Comment (simonlingoogle) [reply pylibs/unittests/test_signals.py:27]: I will do it in a separate PR>

File: pylibs/unittests/test_signals.py (hunk 2 above)
Comment (simonlingoogle) [reply pylibs/unittests/test_signals.py:106]: I think there already is. If not, github will warn us.

File: pylibs/unittests/test_signals.py (hunk 1 above)
Comment (simonlingoogle) [reply pylibs/unittests/test_signals.py:29]: fixed

**Pull request merged**